content/03_CreatingObservableSequences.md

```diff
@@ -0,0 +1,1274 @@
+Although `MySequenceOfNumbers` is technically a correct implementation of `IObservable<int>`, it is a little too simple to be useful. For one thing, we typically use Rx when there are events of interest, but this is not really reactive at all—it just produces a fixed set of numbers immediately. Moreover, the implementation is blocking—it doesn't even return from `Subscribe` until after it has finished producing all of its values. This example illustrates the basics of how a source provides events to a subscriber, but if we just want to represent a predetermined sequence of numbers, we might as well use an `IEnumerable<T>` implementation such as `List<T>` or an array.
+
+## Representing Filesystem Events in Rx
+
+Let's look at something a little more realistic. This is a wrapper around .NET's `FileSystemWatcher`, presenting filesystem change notifications as an `IObservable<FileSystemEventArgs>`. (Note: this is not necessarily the best design for an Rx `FileSystemWatcher` wrapper. The watcher provides events for several different types of change, and one of them, `Renamed`, provides details as a `RenamedEventArgs`. This derives from `FileSystemEventArgs` so collapsing everything down to a single event stream does work, but this would be inconvenient for applications that wanted access to the details of rename events. A more serious design problem is that this is incapable of reporting more than one event from `FileSystemWatcher.Error`. Such errors might be transient and recoverable, in which case an application might want to continue operating, but since this class chooses to represent everything with a single `IObservable<T>`, it reports errors by invoking the observer's `OnError`, at which point the rules of Rx oblige us to stop. It would be possible to work around this with Rx's `Retry` operator, which can automatically resubscribe after an error, but it might be better to offer a separate `IObservable<ErrorEventArgs>` so that we can report errors in a non-terminating way. However, the additional complication of that won't always be warranted—the simplicity of this design means it will be a good fit for some applications. As is often the way with software design, there isn't a one-size-fits-all approach.)
+
+```cs
+// Represents filesystem changes as an Rx observable sequence.
+// NOTE: this is an oversimplified example for illustration purposes.
+//       It does not handle multiple subscribers efficiently, it does not
+//       use IScheduler, and it stops immediately after the first error.
+public class RxFsEvents : IObservable<FileSystemEventArgs>
+{
+    private readonly string folder;
+
+    public RxFsEvents(string folder)
+    {
+        this.folder = folder;
+    }
+
+    public IDisposable Subscribe(IObserver<FileSystemEventArgs> observer)
+    {
+        // Inefficient if we get multiple subscribers.
+        FileSystemWatcher watcher = new(this.folder);
+
+        // FileSystemWatcher's documentation says nothing about which thread
```

Comment:
Something I learned from comments on our first book was that people use code samples without reading the surround text.

We received feedback about the fact that one of the code examples didn't work. The paragraph immediately before the example explained that it was showing something that you shouldn't do. So did the paragraph afterwards.

Since this is:

* one of the larger examples in the book
* a self-contained and apparently complete unit of functionality
* something some people might look at and think is useful

I felt that the comments were a good idea, because this code seemed likely to be transplanted